src/pkg/egress/syslog/https_batch_test.go

```diff
@@ -0,0 +1,180 @@
+		Expect(triggered).To(Equal(1))
+		//expect timer to not be able to be retriggered
+		timer.Trigger()
+		Expect(timer.Running()).To(BeFalse())
+		Expect(triggered).To(Equal(1))
+	})
+})
+
+func trigger() {
+	triggered += 1
+}
+
+var _ = Describe("HTTPS_batch_testing", func() {
+	var (
+		netConf          syslog.NetworkTimeoutConfig
+		skipSSLTLSConfig = &tls.Config{
+			InsecureSkipVerify: true, //nolint:gosec
+		}
+		c      = syslog.NewConverter()
+		drain  *SpyDrain
+		b      *syslog.URLBinding
+		writer egress.WriteCloser
+	)
+	string_to_1024_chars += string_to_1024_chars
+
+	BeforeEach(func() {
+		drain = newBatchMockDrain(200)
+		b = buildURLBinding(
+			drain.URL,
+			"test-app-id",
+			"test-hostname",
+		)
+		writer = syslog.NewHTTPSBatchWriter(
+			b,
+			netConf,
+			skipSSLTLSConfig,
+			&metricsHelpers.SpyMetric{},
+			c,
+		)
+	})
+
+	It("testing simple appending of one log", func() {
+		env1 := buildLogEnvelope("APP", "1", "message 1", loggregator_v2.Log_OUT)
+		Expect(writer.Write(env1)).To(Succeed())
+		env2 := buildLogEnvelope("APP", "2", "message 2", loggregator_v2.Log_OUT)
+		Expect(writer.Write(env2)).To(Succeed())
+		time.Sleep(2 * time.Second)
+
+		Expect(drain.messages).To(HaveLen(2))
+		expected := &rfc5424.Message{
+			AppName:   "test-app-id",
+			Hostname:  "test-hostname",
+			Priority:  rfc5424.Priority(14),
+			ProcessID: "[APP/1]",
+			Message:   []byte("message 1\n"),
+		}
+		Expect(drain.messages[0].AppName).To(Equal(expected.AppName))
+		Expect(drain.messages[0].Hostname).To(Equal(expected.Hostname))
+		Expect(drain.messages[0].Priority).To(BeEquivalentTo(expected.Priority))
+		Expect(drain.messages[0].ProcessID).To(Equal(expected.ProcessID))
+		Expect(drain.messages[0].Message).To(Equal(expected.Message))
+		expected = &rfc5424.Message{
+			AppName:   "test-app-id",
+			Hostname:  "test-hostname",
+			Priority:  rfc5424.Priority(14),
+			ProcessID: "[APP/2]",
+			Message:   []byte("message 2\n"),
+		}
+		Expect(drain.messages[1].AppName).To(Equal(expected.AppName))
+		Expect(drain.messages[1].Hostname).To(Equal(expected.Hostname))
+		Expect(drain.messages[1].Priority).To(BeEquivalentTo(expected.Priority))
+		Expect(drain.messages[1].ProcessID).To(Equal(expected.ProcessID))
+		Expect(drain.messages[1].Message).To(Equal(expected.Message))
+	})
+
+	It("test early dispatch on high message load", func() {
+		env1 := buildLogEnvelope("APP", "1", "string to get log to 1024 characters:"+string_to_1024_chars, loggregator_v2.Log_OUT)
+		for i := 0; i < 300; i++ {
+			writer.Write(env1)
+		}
+		Expect(drain.messages).To(HaveLen(256))
+	})
+
+	It("test batch dispatching with all logs in a given timeframe", func() {
+		env1 := buildLogEnvelope("APP", "1", "string to get log to 1024 characters:"+string_to_1024_chars, loggregator_v2.Log_OUT)
+		for i := 0; i < 10; i++ {
+			writer.Write(env1)
+			time.Sleep(99 * time.Millisecond)
+		}
+		time.Sleep(100 * time.Millisecond)
+		Expect(drain.messages).To(HaveLen(10))
```

Comment:
In general in these tests can we use `Eventually` rather than relying on timing?

Reply:
These are not final yet.
this test especially is set to test for the time window being exactly 1s, like defined in the class, and not something else.
If I reread it right now, it does not exactly test that, but an eventually would not test, if the timings defined would be adhered to.